[CLEANUP beta] #16391 Cleaning up the test output

Draggha opened the pull request:
This resolves some Quest Tasks of Issue #16391:

```
Testing paused. Use resumeTest() to continue.
```
--> I switched the `console.info` with a call to `ember-debug.info` and then used the `setDebugFunction`-helper to set it to a NOOP and reset its behavior after each test accordingly.


```
WARNING: Library "magic" is already registered with Ember.
```
--> I overwrote the debug function "warn", so that the test case could register the same lib a second time without throwing an error.

Draggha committed:
[CLEANUP beta] #16391 Cleaning up the test output
solves the task "WARNING: Library "magic" is already registered with Ember."

rwjblue commented on packages/ember-testing/lib/helpers/pause_test.js:
```diff
@@ -57,9 +57,6 @@ export function resumeTest() {
 */
 export function pauseTest() {
-  // eslint-disable-next-line no-console
-  console.info('Testing paused. Use `resumeTest()` to continue.');
```
We don't want to remove this, it is intentionally logging to the console to let developers know how they can resume. Please revert this removal...

Instead of removing this, we should override `console.info` in the specific tests (and confirm it is called properly).

Draggha replied:
Thanks Robert! I figured that outright deleting this part could have been a bit too much.
Overwriting `console.info` should be possible, but what I fear is that the test code which relies on functions such as `pauseTest` will rely more and more on such workarounds. Every occurrence of console.info, console.warn and the like would need to be patched (now and in future tests).

chancancode replied:
It might be a good idea to indirect these through a `@ember/debug` function (which makes it easier to swap out), what do you think @rwjblue?

rwjblue reviewed (changes requested):
The fix for the library registration warning looks great, but the other one needs some tweaks before landing.

chancancode commented on packages/ember-metal/tests/libraries_test.js:
```diff
@@ -33,6 +34,8 @@ moduleFor('Libraries registry', class extends AbstractTestCase {
     assert.expect(3);
 
+    // overwrite warn to supress the double registration warning (see https://github.com/emberjs/ember.js/issues/16391)
+    setDebugFunction('warn', noop);
```
@rwjblue do we need to set it back after the test, or is that automatically handled by the test harness?

rwjblue replied:
It’s already resetting in the afterEach of this module

Draggha replied:
Yup, that's why I didn't reset it afterwards. :)

Draggha committed:
[CLEANUP beta] #16391 Cleaning up the test output
solves the task "Testing paused. Use resumeTest() to continue."

Draggha commented:
Thanks to both of you for your patience and support. I updated the first task.

```
Testing paused. Use resumeTest() to continue.
```

I also updated the opening post to reflect the current state of this PR.

How would you like to proceed? Would you like to create a branch where this PR would land or will this land inm master? And would you like me open more PRs for other tasks from the issue or should I update this branch? I didn't find anything about your git workflow in the CONTRIBUTING guide. Maybe there is another document somewhere?

rwjblue reviewed (approved):
Thank you!!
